atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderAppender.java

```diff
@@ -363,6 +342,28 @@ public void close() {
                 new RaftException.ProtocolException("Failed to reach consensus")));
   }
 
+  private void tryToReplicateSnapshot(final RaftMemberContext member) {
+    final var optSnapshot = raft.getPersistedSnapshotStore().getLatestSnapshot();
+
+    if (optSnapshot.isPresent()
+        && member.getSnapshotIndex() < optSnapshot.get().getIndex()
+        && optSnapshot.get().getIndex() >= member.getLogReader().getCurrentIndex()) {
+      if (!member.canInstall()) {
+        return;
+      }
+
+      final var persistedSnapshot = optSnapshot.get();
+      log.debug(
+          "Replicating snapshot {} to {}",
+          persistedSnapshot.getIndex(),
+          member.getMember().memberId());
+      buildInstallRequest(member, persistedSnapshot)
+          .ifPresent(installRequest -> sendInstallRequest(member, installRequest));
```

Comment:
What happens when we failed to open the reader and return here empty? No install request is send. Is this an issue?

Reply:
Ya. Then no install request is send. But will be retried in the next heartbeat/appendEntry turn. In worst case, it will be retried for ever if there is something seriously wrong with the snapshot. The other option is to throw an exception, but this will cause the leader to fail and go to INACTIVE phase.